### PR TITLE
Feature/4540/commit id fix

### PR DESCRIPTION
**Pull request by NatalieEO**
**Description**
I messed up my original PR for #4540 while addressing feedback. Added test.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [ ] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [ ] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [ ] Check the Snyk dashboard to ensure you are not introducing new high/critical vulnerabilities
- [ ] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [ ] Do not serve user-uploaded binary images through the Dockstore API
- [ ] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [ ] Do not create cookies, although this may change in the future

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java [hunk 1]
```diff
@@ -534,20 +532,19 @@ public GHRepository getRepository(String repositoryId) {
...
+            sha = repository.getTagObject(ref.getObject().getSha()).getObject().getSha();
         } else if (ref.getObject().getType().equals("branch")) {
             GHBranch branch = repository.getBranch(refName);
             sha = branch.getSHA1();
         } else {
             // I'm not sure when this would happen.
             LOG.error("Unsupported GitHub reference object. Unable to find commit ID for type: " + ref.getObject().getType());
             // This is probably wrong, but we should mimic the behaviour from before since this is a hotfix.
-            sha = ref.getObject().getSha();
+            // sha = ref.getObject().getSha();
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:547]: Channeling Denis, don't leave in commented code unless you expect it to come back.

But don't do another commit for this alone; only address if you have to make other changes to the PR.

**Review by coverbeck — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java [hunk 2]
```diff
@@ -534,20 +532,19 @@ public GHRepository getRepository(String repositoryId) {
     // The documentation doesn't list the possibilities https://github-api.kohsuke.org/apidocs/org/kohsuke/github/GHRef.GHObject.html#getType(),
     // but I'll assume it mirrors the 4 Git types: blobs, trees, commits, and tags.
     private String getCommitSHA(GHRef ref, GHRepository repository, String refName) throws IOException {
-        String sha = null;
+        String sha = ref.getObject().getSha();
         if ("commit".equals(ref.getObject().getType())) {
             sha = ref.getObject().getSha();
```
Comment (kathy-t) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:537]: This line is assigning the same value to `sha` as line 535, so maybe this if statement isn't needed?

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java [hunk 3]
```diff
@@ -534,20 +532,19 @@ public GHRepository getRepository(String repositoryId) {
...
         } else if (ref.getObject().getType().equals("tag")) {
-            GHTagObject tagObject = repository.getTagObject(sha);
-            sha = tagObject.getObject().getSha();
+            sha = repository.getTagObject(ref.getObject().getSha()).getObject().getSha();
```
Comment (kathy-t) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:539]: You could do `repository.getTagObject(sha)` instead of `repository.getTagObject(ref.getObject().getSha())` since `sha` already has that value

**Review by kathy-t — APPROVED** [review]

**Review by garyluu — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java [hunk 4]
```diff
@@ -534,20 +532,18 @@ public GHRepository getRepository(String repositoryId) {
...
     private String getCommitSHA(GHRef ref, GHRepository repository, String refName) throws IOException {
-        String sha = null;
-        if ("commit".equals(ref.getObject().getType())) {
-            sha = ref.getObject().getSha();
-        } else if (ref.getObject().getType().equals("tag")) {
-            GHTagObject tagObject = repository.getTagObject(sha);
-            sha = tagObject.getObject().getSha();
+        // Sha value if commit
+        String sha = ref.getObject().getSha();
+        if (ref.getObject().getType().equals("tag")) {
```
Comment (svonworl) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:537]: Some minor things that might improve the readability and safety of this section:
1. Eliminate duplicate `ref.getObject().getType()` invocations.  Call once and store in local var.  Named `type`?
2. Use `Objects.equal` to compare strings, for graceful null handling.
3. Declare `sha` but don't immediately assign, set value in each conditional block of code.  That way, the compiler will let us know if one of the blocks didn't set a value, and it takes a bit less effort to figure out how it works, imho.

**Review by svonworl — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java (hunk 4 above)
Comment (coverbeck) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:537]: > 2\. Use `Objects.equal` to compare strings, for graceful null handling.

Or put the string first, e.g., `"tag".equals...`.

But I think we had existing code like this, and I want to limit the scope of the change for the hotfix, just in case... So be careful...

**Review by coverbeck — APPROVED** [review]

**Review by svonworl — APPROVED** [review]
